validator/coordinator/src/main/java/tech/pegasys/teku/validator/coordinator/performance/RecentChainDataPerformanceTracker.java

```diff
@@ -0,0 +1,261 @@
+    this.statusLogger = statusLogger;
+  }
+
+  @Override
+  public void start(UInt64 nodeStartSlot) {
+    this.nodeStartEpoch = Optional.of(compute_epoch_at_slot(nodeStartSlot));
+  }
+
+  @Override
+  public void onSlot(UInt64 slot) {
+    if (nodeStartEpoch.isEmpty()) {
+      return;
+    }
+
+    UInt64 currentEpoch = compute_epoch_at_slot(slot);
+    if (!compute_start_slot_at_epoch(currentEpoch).equals(slot)) {
+      return;
+    }
+
+    // Output attestation performance information for current epoch - 2 since attestations can be
+    // included in both the epoch they were produced in or in the one following.
+    if (currentEpoch.isGreaterThanOrEqualTo(nodeStartEpoch.get().plus(UInt64.valueOf(2)))) {
+      statusLogger.performance(
+          getAttestationPerformanceForEpoch(currentEpoch, currentEpoch.minus(UInt64.valueOf(2)))
+              .toString());
+    }
+
+    // Output block performance information for the past BLOCK_PERFORMANCE_INTERVAL epochs
+    if (currentEpoch.isGreaterThanOrEqualTo(BLOCK_PERFORMANCE_EVALUATION_INTERVAL)) {
+      if (currentEpoch.mod(BLOCK_PERFORMANCE_EVALUATION_INTERVAL).equals(UInt64.ZERO)) {
+        statusLogger.performance(
+            getBlockPerformanceForEpochs(
+                    currentEpoch.minus(BLOCK_PERFORMANCE_EVALUATION_INTERVAL), currentEpoch)
+                .toString());
+      }
+    }
+
+    UInt64 epoch =
+        currentEpoch
+            .minus(BLOCK_PERFORMANCE_EVALUATION_INTERVAL)
+            .min(currentEpoch.minus(UInt64.valueOf(2)));
+    clearReduntantSavedSentObjects(epoch);
+  }
+
+  private BlockPerformance getBlockPerformanceForEpochs(
+      UInt64 startEpochInclusive, UInt64 endEpochExclusive) {
+    Set<BeaconBlock> blockInEpoch = getBlocksInEpochs(startEpochInclusive, endEpochExclusive);
+    List<SignedBeaconBlock> sentBlocks =
+        sentBlocksByEpoch.subMap(startEpochInclusive, true, endEpochExclusive, false).values()
+            .stream()
+            .flatMap(Collection::stream)
+            .collect(Collectors.toList());
+    long numberOfIncludedBlocks =
+        sentBlocks.stream()
+            .map(SignedBeaconBlock::getMessage)
+            .filter(blockInEpoch::contains)
+            .count();
+
+    return new BlockPerformance((int) numberOfIncludedBlocks, sentBlocks.size());
+  }
+
+  private AttestationPerformance getAttestationPerformanceForEpoch(
+      UInt64 currentEpoch, UInt64 analyzedEpoch) {
+    checkArgument(
+        analyzedEpoch.isLessThanOrEqualTo(currentEpoch.minus(UInt64.valueOf(2))),
+        "Epoch to analyze attestation performance must be at least 2 epochs less than the current epoch");
+    // Attestations can be included in either the epoch they were produced in or in
+    // the following epoch. Thus, the most recent epoch for which we can evaluate attestation
+    // performance is current epoch - 2.
+    UInt64 analysisRangeEndEpoch = analyzedEpoch.plus(UInt64.valueOf(2));
```

Comment:
Might be good to have the 2-epoch value saved as a constant since it's used throughout
```suggestion
    UInt64 analysisRangeEndEpoch = analyzedEpoch.plus(ATTESTATION_INCLUSION_RANGE);
```

Reply:
Makes sense. Done.